sleepecg/heartbeat_detection.py

```diff
@@ -228,3 +259,395 @@ def rri_similarity(
         scipy.stats.spearmanr(det_resampled, ann_resampled)[0],
         np.sqrt(np.mean((det_resampled - ann_resampled)**2)),
     )
+
+
+def _squared_moving_integration_py(x: np.ndarray, window_length: int) -> np.ndarray:
```

Comment:
Right, but it can only be jitted if numba is available, and I think writing another decorator which considers the case that it isn't would not be prettier.

np.convolve takes about 5 times as long as _squared_moving_integration_numba. Using it would make the Numba backend 50% slower than C (instead of 25%). I didn't manage to jit np.convolve :/ I'd keep it as it is a 50% speedup for the Numba backend.